### PR TITLE
Add Sapphire Rapids as architecture

### PR DESCRIPTION
**Pull request by masterleinad**
Support was already added to `clang` in https://reviews.llvm.org/D86503 (which also uses `SPR` as abbreviation).

### CODE REVIEW

**Review by crtrott — APPROVED** [review]

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -57,6 +57,7 @@ DECLARE_AND_CHECK_HOST_ARCH(ICX               "Intel Ice Lake Server CPUs (AVX51
 DECLARE_AND_CHECK_HOST_ARCH(SKX               "Intel Sky Lake Xeon E-class HPC CPUs (AVX512)")
 DECLARE_AND_CHECK_HOST_ARCH(KNC               "Intel Knights Corner Xeon Phi")
 DECLARE_AND_CHECK_HOST_ARCH(KNL               "Intel Knights Landing Xeon Phi")
+DECLARE_AND_CHECK_HOST_ARCH(SPR               "Intel Sapphire Rapids Xeon Server CPUs (AVX512)")
```
Comment (dalg24) [inline cmake/kokkos_arch.cmake:60]: Please confirm we should be saying "server"

Comment (crtrott) [reply cmake/kokkos_arch.cmake:60]: Yeah its fine, its kind of doubled since Xeon in and of itself implies server but its also fine to say it.

**Comment by dalg24** [comment]
Please rebase

**Comment by masterleinad** [comment]
> Please rebase

Done

**Pull request merged**